.github/workflows/build.yml

```diff
@@ -135,11 +135,37 @@ jobs:
           --name galasa-${{ matrix.region.name }}-site \
           --image $IMAGE_NAME
 
+# Deploy the next branch to next.galasa.dev
+  deploy-next:
+    name: Deploy to next.galasa.dev
+    runs-on: ubuntu-latest
+    needs: [ publish-dotdev ]
+    if: github.ref == 'refs/heads/next'
+    steps:
+      - name: Install IBM Cloud CLI
+        run: |
+          curl -fsSL https://clis.cloud.ibm.com/install/linux | sh
+          ibmcloud --version
+          ibmcloud config --check-version=false
+          ibmcloud plugin install -f code-engine
+
+      - name: Authenticate with IBM Cloud CLI
+        run: |
+          ibmcloud login --apikey "${{ secrets.IBM_CLOUD_API_KEY }}" -r "eu-gb" -g ${{ vars.RESOURCE_GROUP }}
+          ibmcloud ce project select --name galasa-eu-gb
+
+      - name: Update Code Engine application
+        run: |
+          ibmcloud ce application update \
+          --name galasa-eu-gb-next \
+          --image $IMAGE_NAME
+
```

Comment:
Because you've added the ref requirement for the crawler, I don't think this needs duplication (and therefore the possibility of getting out of step). `deploy-dotdev` should do the trick. You'll need to:
- make the `ce application update` have have an expression like `${{ (github.ref == 'refs/heads/main') && '' || '-next'  }}` to set the name.
- Adjust the matrix to have an expression - all 3 on `refs/heads/main`, and just 1 otherwise.

Reply:
Changes made as requested, can you please review again